payment-request/PaymentRequestUpdateEvent/updateWith-state-checks-manual.https.html

```diff
@@ -0,0 +1,93 @@
+<!doctype html>
+<meta charset="utf8">
+<link rel="help" href="https://www.w3.org/TR/payment-request/#updatewith()-method">
+<title>
+</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+setup({ explicit_done: true, explicit_timeout: true });
+const validMethod = Object.freeze({ supportedMethods: "basic-card" });
+const validMethods = Object.freeze([validMethod]);
+const validAmount = Object.freeze({ currency: "USD", value: "5.00" });
+const validTotal = Object.freeze({
+  label: "label",
+  amount: validAmount,
+});
+const validShippingOption = Object.freeze({
+  id: "a-shipping-option",
+  label: "A shipping option",
+  amount: validAmount,
+  selected: true,
+});
+const validDetails = Object.freeze({
+  total: validTotal,
+  shippingOptions: [validShippingOption],
+});
+const validOptions = Object.freeze({
+  requestShipping: true,
+});
+
+function getPaymentPromises() {
+  const request = new PaymentRequest(validMethods, validDetails, validOptions);
+  const eventPromise = new Promise(resolve => {
+    request.addEventListener("shippingaddresschange", resolve);
+  });
+  const responsePromise = request.show();
+  return { eventPromise, responsePromise };
+}
+
+function testRequestIsClosed({ textContent: testName }) {
+  promise_test(async t => {
+    const { eventPromise, responsePromise } = getPaymentPromises();
+    const event = await eventPromise;
+    // Set request.[[state]] to closed
+    await event.target.abort();
+    assert_throws(
+      "InvalidStateError",
+      () => {
+        event.updateWith(validDetails);
+      },
+      "request.[[state]] is not interactive, must throw an InvalidStateError."
+    );
+    responsePromise.catch(err => err);
+  }, testName.trim());
+}
+
+function testRequestIsUpdating({ textContent: testName }) {
+  promise_test(async t => {
+    const { eventPromise, responsePromise } = getPaymentPromises();
+    const event = await eventPromise;
+    // Set request.[[updating]] to true
+    event.updateWith(validDetails);
+    assert_throws(
+      "InvalidStateError",
+      () => {
+        event.updateWith(validDetails);
```

Comment:
I guess there is probably a race condition.
We can't make sure event.updateWith() is called before the internal handling of the last updateWith() finishes.
I suggest that using async function to create a promise and don't resolve/reject it to make sure the that PaymentRequest.[[state]] is updating.

async function detailsPromise() {
    return new Promise((resolved, rejected) => {});
}
event.updateWith(detailsPromise);
assert_throws(
    ...
    event.updateWith(validDetails);
    ...
}

Reply:
Good catch!